itou/utils/models.py

```diff
@@ -62,3 +63,23 @@ class PkSupportRemark(AbstractSupportRemark):
 
 class UUIDSupportRemark(AbstractSupportRemark):
     object_id = models.UUIDField()
+
+
+class UniqueConstraintWithErrorCode(models.UniqueConstraint):
+    def __init__(self, *args, validation_error_code, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.validation_error_code = validation_error_code
+
+    def validate(self, *args, **kwargs):
+        try:
+            return super().validate(*args, **kwargs)
+        except ValidationError as ve:
+            if not ve.code:
+                ve.code = self.validation_error_code
+            raise ve
+
+    def deconstruct(self):
+        path, args, kwargs = super().deconstruct()
+        if self.validation_error_code:
+            kwargs["validation_error_code"] = self.validation_error_code
+        return path, self.expressions, kwargs
```

Comment:
Pas la peine de réimplémenter le comportement du parent.
```suggestion
        return path, args, kwargs
```

Reply:
Bien vu :+1: